nhais-adaptor/edifact/outgoing/models/birth/message_birth.py

```diff
@@ -0,0 +1,73 @@
+from edifact.helpers import date_formatter as date_formatter
+from edifact.outgoing.models.address import PatientAddress
+from edifact.outgoing.models.message import MessageSegmentRegistrationDetails, MessageSegmentPatientDetails, Message
+from edifact.outgoing.models.name import PatientName
+from edifact.outgoing.models.segment import Segment
+
+
+class MessageSegmentBirthPatientDetails(MessageSegmentPatientDetails):
+    """
+    A specialisation of the MessageSegmentPatientDetails class for the purpose of a Birth
+    """
+
+    def __init__(self, id_number, name, date_of_birth, gender, address):
+        """
+        :param id_number: OPI official Payment Id (existing NHS Number)
+        :param name: the name of the patient
+        :param date_of_birth: Patients date of birth
+        :param gender: sex of the patient. For an acceptance transaction, reference "G1", this segment is required
+        :param address: the patients address
+        """
+        formatted_date = date_formatter.format_date(date_time=date_of_birth, format_qualifier="102",
+                                                    current_format="%Y-%m-%d")
+        segments = [
+            PatientName(id_number=id_number, name=name),
+            Segment(key="DTM", value=f"329:{formatted_date}:102"),
+            Segment(key="PDI", value=f"{gender}"),
+            PatientAddress(address=address)
+        ]
+        super().__init__()
+        super().add_segments(segments)
+
+
+class MessageSegmentBirthRegistrationDetails(MessageSegmentRegistrationDetails):
+    """
+    A specialisation of the MessageSegmentRegistrationDetails class for the purpose of a Birth
+    """
+
+    def __init__(self, transaction_number, party_id, acceptance_code, acceptance_type, date_time, location):
+        """
+        :param transaction_number: a unique transaction number. NHAIS will reference this in its response
+        :param party_id: GMC National code and the Local GP Code of the patient's GP (separated by “,”).
+        :param acceptance_code: The acceptance code "A" for Acceptance

```

Comment:
If acceptance code and type are always fixed values for a birth, do we need constructor parameters for them here?

Reply:
You are right. These will be fixed values now for a birth registration. Will remove these 👍